### PR TITLE
Merge release-2.10 to main

### PR DESCRIPTION
**Pull request by colega**
In this PR I'm merging the release-2.10 branch to main branch. Please merge this PR using a **merge commit** or by using `tools/release/merge-approved-pr-branch-to-main.sh` script.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -212,7 +212,7 @@ mimir-build-image/$(UPTODATE): mimir-build-image/*
 # All the boiler plate for building golang follows:
 SUDO := $(shell docker info >/dev/null 2>&1 || echo "sudo -E")
 BUILD_IN_CONTAINER ?= true
-LATEST_BUILD_IMAGE_TAG ?= pr5955-facaa1c135
+LATEST_BUILD_IMAGE_TAG ?= pr5960-facaa1c135
```
Comment (pstibrany) [inline Makefile:215]: nit: we don't need this change, #5960 was cherrypick of #5955, but bot decided to rebuild the image.

**Review by pstibrany — APPROVED** [review]
👍

File: Makefile (hunk 1 above)
Comment (pstibrany) [reply Makefile:215]: but don't bother fixing it, the two build images are the same.

Comment (colega) [reply Makefile:215]: Thanks. I saw the changelog with two PRs and just took the latest.  I'll leave this as is.

**Pull request merged**